Added support for ThrowableProblem.getCause

whiskeysierra opened the pull request:
This closes #3

lukasniemeier-zalando commented:
Merge and release?

whiskeysierra commented:
> Merge and release?

What about a new unwritten law that requires at least one :+1: on a pull request before it get's merged?

jhorstmann commented on problem/src/main/java/org/zalando/problem/ThrowableProblem.java:
```diff
@@ -22,9 +22,29 @@
+            return null;
+        }
 
+        checkState(cause instanceof ThrowableProblem, "Expected throwable problem cause, but got %s", cause);
```
Can we check the cause in the constructor instead? That would give a better stacktrace to find the error.

whiskeysierra replied:
> Can we check the cause in the constructor instead? That would give a better stacktrace to find the error.

I changed the parameter type to `ThrowableProblem`

whiskeysierra commented:
@jhorstmann I simplified the `getCause` method (including the cast) and added comment. The correct type is now ensured via the constructor parameter. There was no need for an explicit `instanceof` check anymore.

whiskeysierra commented:
Any objections for merge+release?

lukasniemeier-zalando commented on jackson-datatype-problem/pom.xml:
```diff
@@ -75,6 +75,12 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.slf4j</groupId>
```
:+1:

lukasniemeier-zalando commented:
:+1: go!
